lib/compress/zstd_compress.c

```diff
@@ -3682,23 +3876,23 @@ size_t ZSTD_initCStream_internal(ZSTD_CStream* zcs,
         }
         ZSTD_freeCDict(zcs->cdictLocal);
         zcs->cdictLocal = ZSTD_createCDict_advanced(dict, dictSize,
-                                            zcs->dictContentByRef, zcs->dictMode,
+                                            ZSTD_dlm_byCopy, ZSTD_dm_auto,
```

Comment:
Previously `ZSTD_initCStream()` respected the dict loading parameters set by the new API, but now it doesn't. Is this okay? Without mixing the new API and the old CStream API there is no way to set these parameters, but if someone mixes the two APIs, the behavior changes.

Reply:
I think `ZSTD_initCStream_internal()` is never invoked from the new advanced API,
which goes straight to `ZSTD_resetCStream_internal()` instead.

`ZSTD_initCStream_internal()` is called by : 
- `ZSTD_initCStream_usingCDict_advanced()` : which doesn't use a `dict`, since it refers to a `cdict`
- `ZSTD_initCStream_advanced()` : which can use `dict`, and implies `ZSTD_dm_auto` mode. `ZSTD_dlm_byCopy` loading mode is implied too, just because it's safer and therefore doesn't require the user to pay attention to its `dict` buffer lifetime.
- `ZSTD_initCStream_usingDict()` : which also uses `dict` and implies the `ZSTD_dm_auto` and `ZSTD_dlm_byCopy`.
- `ZSTD_initCStream_srcSize()` : which doesn't use a `dict`.

So, in the 2 valid cases, these parameters are correct.

Still, that's the situation _today_. What if it changes in the future ?
Someone may try to use this interface for some future feature we still don't know about.
So, at a minimum, these implied parameters should be documented.

As forward thinking, they could also become arguments to the function, so they could be modified in the future, but it introduces additional complexity for something which is not useful immediately. So is it worth it ?